otel-installer/docker/docker-install.sh

```diff
@@ -0,0 +1,450 @@
+      exporters: [nop]
+    logs:
+      receivers: [nop]
+      exporters: [nop]
+EOF
+}
+
+create_config() {
+    local config_dir
+    config_dir=$(mktemp -d)
+    
+    if [ -n "$CUSTOM_CONFIG_PATH" ]; then
+        if [ ! -f "$CUSTOM_CONFIG_PATH" ]; then
+            fail "Config file not found: $CUSTOM_CONFIG_PATH"
+        fi
+        cp "$CUSTOM_CONFIG_PATH" "${config_dir}/config.yaml"
+    else
+        get_default_config > "${config_dir}/config.yaml"
+    fi
+    
+    echo "$config_dir"
+}
+
+run_regular_mode() {
+    local default_version
+    default_version=$(get_version)
+    local image_tag="${COLLECTOR_VERSION:-${default_version}}"
+    local image="${COLLECTOR_IMAGE}:${image_tag}"
+    
+    log "Pulling image: ${image}"
+    docker pull "$image"
+    
+    # Create config directory
+    local config_dir
+    config_dir=$(create_config)
+    
+    log "Starting collector container..."
+    
+    local docker_args=(
+        --name "${CONTAINER_NAME}"
+        --restart unless-stopped
+        -e "CORALOGIX_PRIVATE_KEY=${CORALOGIX_PRIVATE_KEY}"
+        -v "${config_dir}/config.yaml:/etc/otelcol-contrib/config.yaml:ro"
+        -p 4317:4317
+        -p 4318:4318
+        -p 13133:13133
+    )
+    
+    if [ -n "${CORALOGIX_DOMAIN:-}" ]; then
+        docker_args+=(-e "CORALOGIX_DOMAIN=${CORALOGIX_DOMAIN}")
+    fi
+    
+    if [ "$DETACHED" = true ]; then
+        docker_args+=(-d)
+    fi
+    
+    # Remove existing container if present
+    docker rm -f "${CONTAINER_NAME}" 2>/dev/null || true
+    
+    docker run "${docker_args[@]}" "$image"
+    
+    if [ "$DETACHED" = true ]; then
+        log "Container started successfully"
+        print_summary
+    fi
+}
+
+get_supervisor_config() {
+    local domain="$1"
+    cat <<EOF
+server:
+  endpoint: "https://ingress.${domain}/opamp/v1"
+  headers:
+    Authorization: "Bearer \${env:CORALOGIX_PRIVATE_KEY}"
+  tls:
+    insecure_skip_verify: true
+
+capabilities:
+  reports_effective_config: true
+  reports_own_metrics: true
+  reports_own_logs: true
+  reports_own_traces: true
+  reports_health: true
+  accepts_remote_config: true
+  reports_remote_config: true
+
+agent:
+  executable: /otelcol-contrib
+  passthrough_logs: true
+
+  description:
+    non_identifying_attributes:
+      service.name: "opentelemetry-collector"
+      cx.agent.type: "docker"
+
+  config_files:
+    - /etc/otelcol-contrib/config.yaml
+
+  args: []
+
+  env:
+    CORALOGIX_PRIVATE_KEY: "\${env:CORALOGIX_PRIVATE_KEY}"
+
+storage:
+  directory: /etc/otelcol-contrib/supervisor-data/
+
+telemetry:
+  logs:
+    level: info
+EOF
+}
+
+run_supervisor_mode() {
+    local default_version
+    default_version=$(get_version)
+    local image_tag="${SUPERVISOR_VERSION:-${default_version}}"
+    local image="${SUPERVISOR_IMAGE}:${image_tag}"
+    
+    log "Pulling image: ${image}"
+    docker pull "$image"
+    
+    # Create config directory
+    local config_dir
+    config_dir=$(mktemp -d)
```

Comment:
These folders are created under the folder pointed in `TMPDIR` environment variable. Between reboots, those folders are removed -if I'm not mistaken- or according to `TMPTIME`. In the second case, it can happen when the service is not running -which I think we should not worry about- but in the first, if the machine is restarted, the collector will not be able to read the configuration.

Reply:
Great observation, @iblancasa. We cannot not use `mktemp` for anything that is not explicitly temporary. For any persistent file we should follow the unix standard and have our own folder inside `/etc`. A suggestion could be `/etc/opampsupervisor`, which is what we use upstream in the systemd service for the Supervisor.